tests/mitmproxy_addons/callback.py

```diff
@@ -71,26 +74,21 @@ def response(self, flow):
                 res_body = flow.response.json()
             except:
                 res_body = None
-            data = json.dumps({
-                "method": flow.request.method,
-                "access_token": flow.request.headers.get("Authorization", "").removeprefix("Bearer "),
-                "url": flow.request.url,
-                "response_code": flow.response.status_code,
-                "request_body": req_body,
-                "response_body": res_body,
-            })
-            request = Request(
-                self.config["callback_url"],
-                headers={"Content-Type": "application/json"},
-                data=data.encode("utf-8"),
-            )
+            print(f'{datetime.now().strftime("%H:%M:%S.%f")} hitting callback for {flow.request.url}')
             try:
-                with urlopen(request, timeout=10) as response:
-                    print(f"callback returned HTTP {response.status}")
-                    return response.read(), response
-            except HTTPError as error:
-                print(f"ERR: callback returned {error.status} {error.reason}")
-            except URLError as error:
-                print(f"ERR: callback returned {error.reason}")
-            except TimeoutError:
-                print(f"ERR: callback request timed out")
+                # use asyncio so we don't block other unrelated requests from being processed
+                async with aiohttp.request(
+                    method="POST",url=self.config["callback_url"], timeout=aiohttp.ClientTimeout(total=10),
+                    headers={"Content-Type": "application/json"},
+                    json={
+                        "method": flow.request.method,
+                        "access_token": flow.request.headers.get("Authorization", "").removeprefix("Bearer "),
+                        "url": flow.request.url,
+                        "response_code": flow.response.status_code,
+                        "request_body": req_body,
+                        "response_body": res_body,
+                    }) as response:
+                    print(f'{datetime.now().strftime("%H:%M:%S.%f")} callback for {flow.request.url} returned HTTP {response.status}')
+                    return
```

Comment:
Is this supposed to be a bare `return`? If so, isn't it useless as we're at the end anyway?

Does anyone do anything with the response that we used to return?

Reply:
It currently does nothing with the response returned, but it will end up modifying it later down the line. The `return` is redundant though, so will remove.